staging/src/slime.io/slime/modules/lazyload/controllers/service_cache.go

```diff
@@ -115,8 +96,97 @@ func newSvcCache(clientSet *kubernetes.Clientset) (*NsSvcCache, *LabelSvcCache,
 			}
 			labelSvcCache.Unlock()
 
+			// add eventSvc ports to portProtocolCache again
+			if ns != r.env.Config.Global.IstioNamespace {
+				portProtocolCache.Lock()
+				for _, port := range service.Spec.Ports {
+					p := port.Port
+					if portProtocolCache.Data[p] == nil {
+						portProtocolCache.Data[p] = make(map[Protocol]uint)
+					}
+					switch getProtocol(port) {
+					case ProtocolHTTP:
+						portProtocolCache.Data[p][ProtocolHTTP]++
+					case ProtocolTCP:
+						portProtocolCache.Data[p][ProtocolTCP]++
+					case ProtocolUnknown:
+						portProtocolCache.Data[p][ProtocolUnknown]++
+					}
+				}
+				portProtocolCache.Unlock()
+			}
+
 		}
 	}()
 
-	return nsSvcCache, labelSvcCache, nil
+	if r.cfg.AutoPort {
+		log.Infof("Port auto management by lazyload is running")
+		go func() {
+			ticker := time.NewTicker(10 * time.Second)
+			for {
+				select {
+				case <-ticker.C:
+					// update wormholePort
+					log.Debugf("got timer event for updating wormholePort")
```

Comment:
the first execution may be too late. we can import an `init-delay` concept

Reply:
add retry ticker for init and retry